apollo-router/src/state_machine.rs

```diff
@@ -320,27 +319,37 @@ impl<FA: RouterSuperServiceFactory> State<FA> {
             Schema::parse_arc(schema_state.clone(), &configuration)
                 .map_err(|e| ServiceCreationError(e.to_string().into()))?,
         );
+        let license_limits = license.get_limits();
+
         // Check the license
-        let report = LicenseEnforcementReport::build(&configuration, &schema);
+        let report = LicenseEnforcementReport::build(&configuration, &schema, license_limits);
 
-        let license_limits = match license {
+        match license {
             LicenseState::Licensed { limits } => {
-                tracing::debug!("A valid Apollo license has been detected.");
-                limits
+                let license_is_restricted = limits
+                    .map(|limits| limits.restricted.unwrap_or(false))
+                    .unwrap_or(false);
+                if report.uses_restricted_features() && license_is_restricted {
+                    tracing::error!(
```

Comment:
I'd like to see messaging delivered as part of the JWT (with defaults in the router). That way we can change what we display from the server side without having to update the router code.